src/main/scala/net/jcazevedo/moultingyaml/YamlPrinter.scala

```diff
@@ -0,0 +1,74 @@
+package net.jcazevedo.moultingyaml
+
+import org.yaml.snakeyaml.{Yaml, DumperOptions}
+
+/**
+  * @author Simone D'Avico (simonedavico@gmail.com)
+  *
+  * Created on 22/02/16.
+  */
+trait OptionsMapping[Flow, Scalar] {
+  def corresponding(flow: FlowStyle): Flow
+  def corresponding(scalar: ScalarStyle): Scalar
+}
+
+sealed trait FlowStyle
+case class Auto() extends FlowStyle
+case class Block() extends FlowStyle
+case class Flow() extends FlowStyle
```

Comment:
I didn't even know case objects where a thing. I will change all the classes to objects.